### DISCUSSION TITLE
Is there support for reading RFID (NFC) and scanning QR Code?

### DISCUSSION THREAD

**Discussion by bangzek** [part-cw/lambdanative#439] [Q&A]
Is there support for reading RFID (NFC) and scanning QR Code?

**Comment by mgorges** [comment]
Dear @bangzek,
There isn't support for NFC, but you are welcome to contribute that code if you are interested in this topic; unfortunately, our team can't do so for the multiple platforms this framework supports at this time. 

For QR code scanning 9108ddd added support for the [QUIRC](https://github.com/dlbeer/quirc) library, which you find in [modules/quirc/quirc.scm](https://github.com/part-cw/lambdanative/blob/74ec19dddf2f2ff787ee70ad677bc13b9dfafc29/modules/quirc/quirc.scm). I don't think there is a demo app for it and it also seems to be missing from the module list in the [Wiki](https://github.com/part-cw/lambdanative/wiki). Usage is straight forward, you pass a PNG file to the function [(png->qrcodes pngfile)](https://github.com/part-cw/lambdanative/blob/74ec19dddf2f2ff787ee70ad677bc13b9dfafc29/modules/quirc/quirc.scm#L140) and get the data back.
